### PR TITLE
Prepare v1.7.0 release

### PR DESCRIPTION
**Pull request by msfjarvis**
Last bunch of QoL cleanups in preparation for the v1.7.0 release. This is gonna sit until #725
is either fixed or removed.

**NOTE**: When this is approved, please do not use the GitHub UI to merge it, it will most certainly invalidate
the tag I pushed alongside. I'll push the branch into master to conclude the merge when its time.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -35,6 +35,8 @@ android {
 
     defaultConfig {
         applicationId versions.packageName
+        versionCode 10701
```
Comment (FabianHenneke) [inline app/build.gradle:38]: So the tag will be on the commit preceding this one?

File: app/src/main/java/com/zeapo/pwdstore/ClipboardService.kt [hunk 2]
```diff
@@ -92,7 +93,7 @@ class ClipboardService : Service() {
                 ClipboardUtils.clearClipboard(clipboardManager, deepClear)
             }
         } else {
-            Timber.tag("ClipboardService").d("Cannot get clipboard manager service")
+            tag("ClipboardService").d { "Cannot get clipboard manager service" }
```
Comment (FabianHenneke) [inline app/src/main/java/com/zeapo/pwdstore/ClipboardService.kt:96]: Timber-kt should detect the tag automatically in Debug mode. If this is all we want, we could remove the `tag(...)` here.

File: app/src/main/java/com/zeapo/pwdstore/PasswordStore.kt [hunk 3]
```diff
@@ -647,32 +653,34 @@ class PasswordStore : AppCompatActivity() {
                     startActivityForResult(intent, BaseGitActivity.REQUEST_CLONE)
                 }
                 REQUEST_CODE_SELECT_FOLDER -> {
-                    Timber.tag(TAG)
-                            .d("Moving passwords to ${data!!.getStringExtra("SELECTED_FOLDER_PATH")}")
-                    Timber.tag(TAG).d(
-                            TextUtils.join(", ", requireNotNull(data.getStringArrayListExtra("Files")))
-                    )
+                    val intentData = data?: return
```
Comment (FabianHenneke) [inline app/src/main/java/com/zeapo/pwdstore/PasswordStore.kt:656]:
```suggestion
                    val intentData = data ?: return
```

File: app/src/main/java/com/zeapo/pwdstore/git/GitConfigActivity.kt [hunk 4]
```diff
@@ -50,12 +51,12 @@ class GitConfigActivity : BaseGitActivity() {
...
-                val editor = settings.edit()
-                editor.putString("git_config_user_email", email)
-                editor.putString("git_config_user_name", name)
-                PasswordRepository.setUserName(name)
-                PasswordRepository.setUserEmail(email)
-                editor.apply()
+                settings.edit {
+                    putString("git_config_user_email", email)
+                    putString("git_config_user_name", name)
+                    PasswordRepository.setUserName(name)
```
Comment (FabianHenneke) [inline app/src/main/java/com/zeapo/pwdstore/git/GitConfigActivity.kt:57]: Should these be moved out of the settings.edit { ... } block?

File: app/build.gradle (hunk 1 above)
Comment (msfjarvis) [reply app/build.gradle:38]: Yep, the tag is on f0923250dcc2527da3ae96a0386e02edab24e381

**Comment by msfjarvis** [comment]
Addressed the review comments and force pushed the v1.7.0 tag to 94607e8

**Comment by FabianHenneke** [comment]
What is left to be done about #725?

**Comment by msfjarvis** [comment]
> What is left to be done about #725?

HTTPS password is being persisted into `EncryptedSharedPreferences` but getting cleared for some reason at a later point in time, without any Git operation failing. I just noticed that going into `GitServerConfigActivity` and back with no changes or pressing save cleared the saved password for me.

**Comment by msfjarvis** [comment]
Nvm I just fixed it. 2 days of pulling my hair and rubber ducky is how I find the problem...

**Comment by msfjarvis** [comment]
We should be good to go here.

**Comment by FabianHenneke** [comment]
> We should be good to go here.

Do we have the Autofill onboarding UI ready for this release? If not, a simple dialog pointing to the settings entry on first launch after update would probably do the trick.

Otherwise this is good to go, yes.

**Comment by msfjarvis** [comment]
> > We should be good to go here.
> 
> Do we have the Autofill onboarding UI ready for this release? If not, a simple dialog pointing to the settings entry on first launch after update would probably do the trick.

Never really got around to it, will add the dialog now. Do we want to show it to everybody or just the people on Oreo+ who have the pre-Oreo autofill enabled? I feel the latter is a bit more logical. Dialog takes users to settings so they can disable the service, then they can come back and enable the right one.

**Comment by FabianHenneke** [comment]
I've never used pre-Oreo Autofill and don't know whether it had issues post-Oreo Autofill doesn't have and that could have turned people off. Depending on whether that's the case or not, a general announcement dialog could make sense.

**Comment by msfjarvis** [comment]
> I've never used pre-Oreo Autofill and don't know whether it had issues post-Oreo Autofill doesn't have and that could have turned people off. Depending on whether that's the case or not, a general announcement dialog could make sense.

That does make sense, yeah. Will insert the commit in this PR.

**Comment by msfjarvis** [comment]
> I've never used pre-Oreo Autofill and don't know whether it had issues post-Oreo Autofill doesn't have and that could have turned people off. Depending on whether that's the case or not, a general announcement dialog could make sense.

Pushed as c57d34405535

**Comment by FabianHenneke** [comment]
> > I've never used pre-Oreo Autofill and don't know whether it had issues post-Oreo Autofill doesn't have and that could have turned people off. Depending on whether that's the case or not, a general announcement dialog could make sense.
> 
> Pushed as c57d34405535

While I generally like change, it does mean that most people will never see the actual onboarding dialog with browser compatibilities listed. Maybe we can launch that when they tap ok instead of directly taking them into Android settings?

**Comment by msfjarvis** [comment]
> > > I've never used pre-Oreo Autofill and don't know whether it had issues post-Oreo Autofill doesn't have and that could have turned people off. Depending on whether that's the case or not, a general announcement dialog could make sense.
> > 
> > 
> > Pushed as [c57d344](https://github.com/android-password-store/Android-Password-Store/commit/c57d34405535a23d3d3f09b710d3e4ddeeb4d8fd)
> 
> While I generally like change, it does mean that most people will never see the actual onboarding dialog with browser compatibilities listed. Maybe we can launch that when they tap ok instead of directly taking them into Android settings?

I was concerned about that as well. I'll add the necessary bits from `UserPreference`.
screenshot-20200421-183349

**Pull request merged**